core/src/main/java/org/apache/accumulo/fate/util/Retry.java

```diff
@@ -147,7 +170,12 @@ public long retriesCompleted() {
 
   public void waitForNextAttempt() throws InterruptedException {
     log.debug("Sleeping for {}ms before retrying operation", currentWait);
-    sleep(currentWait);
+    if (!firstRetry)
+      this.waitIncrement = (long) (Math.ceil(this.waitFactor * Double.valueOf(this.waitIncrement)));
```

Comment:
You don't need `Double.valueOf`... it's already a double. This will merely create a boxed object version, which immediately gets unboxed to do the multiplication.

Also, if you use my strategy above on each round, to vary the waitFactor slightly, you don't need to do anything special on the firstRetry.